src/components/ColorPicker.tsx

```diff
@@ -391,34 +366,8 @@ export const ColorPicker = ({
   appState: AppState;
 }) => {
   const pickerButton = React.useRef<HTMLButtonElement>(null);
-  const [customColors, setCustomColors] = React.useState<string[]>([]);
-  useEffect(() => {
-    if (!isActive) {
-      return;
-    }
-    if (type !== "canvasBackground") {
-      setCustomColors(getCustomColors(elements, type));
-    }
-  }, [type, elements, isActive]);
```

Comment:
moved inside `Picker` compo, and I'm also initializing it once on mount. This has the added benefit that the custom color doesn't disappear if you switch to a different color (and it's the only element on canvas using that color) — that way you can still switch back to it before you close the popup.